core/src/main/java/jenkins/util/UrlHelper.java

```diff
@@ -23,36 +23,40 @@
  */
 package jenkins.util;
 
+import jenkins.org.apache.commons.validator.routines.DomainValidator;
 import jenkins.org.apache.commons.validator.routines.UrlValidator;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
 
-import java.net.MalformedURLException;
-import java.net.URL;
-
 @Restricted(NoExternalUse.class)
 public class UrlHelper {
+    /**
+     * Authorize the _ and - characters, avoid them to be first or last
```

Comment:
Might be more clear as `Allow the _ and - characters in domains as long as they are not the first or last character`. That said, `\w` includes `_` so I think the given regex will allow `_` as the first character.

Reply:
Comment is lying, was expected to be only `.` and `-`, will change it, thank you 👍